pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md

```diff
@@ -0,0 +1,209 @@
+
+When you register your Target, the dictionary key is the named property a developer uses to access that Target.
+In the Page Builder example, the dictionary key for the rich content renderers Target that Venia UI declares is `richContentRenderers`.
+
+The value associated with the key is a variant of a Tapable hook created using one of the class constructors available under the TargetProvider object's `types` property.
+Each class constructor accepts an optional array of strings that represent the list of arguments passed to the [intercept function][].
+
+The TargetProvider object exposes the same classes as the [Tapable][] package except their names do not end with 'Hook'.
+This is intentional to avoid confusion with the concept of React Hooks.
+For example, the `SyncWaterfallHook` class from Tapable is `SyncWaterfall` under the TargetProvider object's `types` property.
+
+See the [Hook types][] section of the Tapable documentation to learn about the available hook types.
+
+### Example declare file content
+
+```js
+module.exports = targets => {
+  targets.declare({
+    perfReport: new targets.types.AsyncParallel(['report'])
+    socialIcons: new targets.types.SyncWaterfall(['iconlist'])
+  });
+}
+```
+
+The example provided declares two Targets that this project or other modules can intercept, a `perfReport` Target and a `socialIcons` Target.
+Intercept files can access these Targets using `targets.of(<package name>).perReport` and `targets.of(<package name>).socialIcons` respectively.
+
+The `perfReport` key maps to a Target type that runs its interceptors asynchronously and in parallel.
+This is appropriate for logging and monitoring interceptors that do not affect functionality.
+
+The `socialIcons` key maps to a Target type that runs its interceptors synchronously and in subscription order, which passes its return values as arguments to the next interceptor.
+This is appropriate for customizations that must happen in a predictable order.
+
+Both targets passes in a single parameter to their intercept functions, a `report` argument and an `iconlist` argument.
+
+## Intercept files
+
+The _intercept file_ describes the targets you want to intercept and define or extend.
+During build time, the framework looks for this file using the value for `pwa-studio.targets.intercept` in your project's `package.json` file.
+These files execute after all the declare files register their Targets.
+
+Your intercept files must export a function that accepts a TargetProvider object as a parameter.
+This object gives you access to all available Targets in your project that let you make customizations, gather data, change the build itself, or develop and call your own declared targets.
+The TargetProvider object provides an `of()` function and an `own` property to access Targets in other packages or Targets within its own module respectively.
+
+For more information on how intercept files work, see the tutorial on how to [Intercept a Target][].
```

Comment:
Even a simple example here would be great

Reply:
Added in 84ef809